src/Uno.Toolkit.UI/Controls/TabBar/TabBarItem.Properties.cs

```diff
@@ -29,6 +29,28 @@ partial class TabBarItem
 			DependencyProperty.Register(nameof(Icon), typeof(IconElement), typeof(TabBarItem), new PropertyMetadata(null, OnPropertyChanged));
 		#endregion
 
+		#region BadgeVisibility
+		public Visibility BadgeVisibility
+		{
+			get { return (Visibility)GetValue(BadgeVisibilityProperty); }
+			set { SetValue(BadgeVisibilityProperty, value); }
+		}
+
+		public static readonly DependencyProperty BadgeVisibilityProperty =
+			DependencyProperty.Register("BadgeVisibility", typeof(Visibility), typeof(TabBarItem), new PropertyMetadata(Visibility.Collapsed, OnPropertyChanged));
+		#endregion
+
+		#region BadgeValue
+		public string? BadgeValue
+		{
+			get { return (string)GetValue(BadgeValueProperty); }
+			set { SetValue(BadgeValueProperty, value); }
+		}
+
+		public static readonly DependencyProperty BadgeValueProperty =
+			DependencyProperty.Register("BadgeValue", typeof(string), typeof(TabBarItem), new PropertyMetadata(default(string?), OnPropertyChanged));
+		#endregion
```

Comment:
Is there any nice way of just keeping this as a single DependencyProperty. Like just Badge? Something like [this](https://github.com/unoplatform/Uno.Themes/blob/0b06cd43e2657c13055901ac60bd60889d46a80b/src/samples/Uno.Material.Samples/Uno.Material.Samples.Shared/Content/Controls/BottomNavigationBarSamplePage.xaml#L72)?

Reply:
yeah, MaterialEmptyToCollapsed converter

Reply:
but i would keep the property, and even upgrade it to an enum: None, NotificationDot, TextBadge

Reply:
If we just had a `string? Badge` we could have it Collapsed if null, dot if string.Empty, and full badge if has text.

Ugh kinda weird to have to set `.Badge=""` in order for it to display... Maybe not :/

Reply:
@kazo0 / @Xiaoy312  I was inspired by the InfoBadge when I made that choice at first 
(cf. https://docs.microsoft.com/en-us/windows/apps/design/controls/info-badge#dot)
**With InfoBadge:**
- When the control is not there, there is no badge.
- When the control is there but there is no `Value` property set, it will be the dot version
![image](https://user-images.githubusercontent.com/16295702/187232667-d6a5e33f-7a06-423a-8ab5-bee405a0f01a.png)
- When the control is there and there is the `Value` property set, it will be the numeric version
![image](https://user-images.githubusercontent.com/16295702/187232994-6cd2e65c-5c32-48e6-961d-a2beb90f0c0a.png)

Since in our case, it's not a badge custom control per se, that is why I added the `BadgeVisibility`, and if no `BadgeValue` is set it will be the dot version otherwise it will be the numeric version displaying the value.
It was a simple solution without complicated logic for now, and that is using converters we already have in Themes.

I thought of using an enum at first but we will need to have another property outside of the enum to specify/bind the numeric value at the end, so two properties are also needed here at the end with more logic to add to deal with the enum values. Or am I mistaken?

Reply:
i think visibility+value is the best strategy without a custom control
having a single null/empty/value is just not intuitive at first glance

Reply:
I agree with you @Xiaoy312 
So do we keep it like that for now @kazo0 or should I change something else?
(I will make the proper changes to the indentation of the xaml also)

Reply:
Thanks to both of you for your feedback, really appreciated :) !